packages/devtools_app/lib/src/flutter/navigation.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../ui/theme.dart' as devtools_theme;
+
+/// Returns [routeName] with [queryParameters] appended as [Uri] query
+/// parameters.
+///
+/// If not overridden, this will preserve the theme parameter across
+/// navigations.
+///
+///
+/// This will fail because we can't determine what the theme value for the app
+/// is.
+String routeNameWithQueryParams(BuildContext context, String routeName,
+    [Map<String, String> queryParameters]) {
+  final newQueryParams =
+      queryParameters == null ? null : Map.of(queryParameters);
+
+  String previousQuery;
+  if (context == null) {
+    // We allow null context values to make easy pure-VM tests.
+    previousQuery = '';
+  } else {
+    previousQuery = ModalRoute.of(context).settings.name;
+    // When this function is invoked from an unnamed context,
+    // infer from the global theme configuration.
+    previousQuery ??= _inferThemeParameter;
+  }
+
+  final previousQueryParams = Uri.parse(previousQuery).queryParameters;
+  // Preserve the theme across app-triggered navigations.
+  if (newQueryParams != null &&
+      !newQueryParams.containsKey('theme') &&
+      previousQueryParams.containsKey('theme')) {
+    newQueryParams['theme'] = previousQueryParams['theme'];
```

Comment:
rather than skipping based on whether the params are empty, skip based on whether the value is the default. For example, always omit 'theme=light' as it isn't needed.

Reply:
Done.